make_prg/from_msa/prg_builder.py

```diff
@@ -103,94 +109,128 @@ def get_sub_alignment_by_list_id(
             sub_alignment = sub_alignment[:, interval[0] : interval[1] + 1]
         return sub_alignment
 
+    @classmethod
+    def skip_clustering(
+        cls,
+        interval,
+        nesting_level: int,
+        max_nesting: int,
+        min_match_length: int,
+        alignment: MSA,
+    ) -> bool:
+        max_nesting_reached = nesting_level >= max_nesting
+        if max_nesting_reached:
+            return True
+
+        small_interval = interval.stop - interval.start <= min_match_length
+        if small_interval:
+            return True
+
+        sub_alignment = alignment[:, interval.start : interval.stop + 1]
+        num_unique_nongapped = count(
+            remove_duplicates(map(ungap, get_alignment_seqs(sub_alignment)))
+        )
+        if num_unique_nongapped <= 2:
+            return True
+
+        num_unique_gapped = count(remove_duplicates(get_alignment_seqs(sub_alignment)))
+        assert num_unique_nongapped <= num_unique_gapped
```

Comment:
why?

Reply:
Given a sequence in alphabet `[ATCG]`, say you can add gaps to it anywhere `-`. The set of sequences in the alphabet `[ATCG-]` is a superset of that in `[ATCG]`, so that assertion must hold

Reply:
It's just an extra way to check the counting and ungapping are working fine